atc/api/resourceserver/clear_resource_cache.go

```diff
@@ -0,0 +1,90 @@
+package resourceserver
+
+import (
+	"code.cloudfoundry.org/lager"
+	"encoding/json"
+	"fmt"
+	"github.com/concourse/concourse/atc"
+	"github.com/google/jsonapi"
+	"io"
+	"net/http"
+
+	"github.com/concourse/concourse/atc/db"
+)
+
+func (s *Server) ClearResourceCache(pipeline db.Pipeline) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		logger := s.logger.Session("clear-resource-cache")
+		resourceName := r.FormValue(":resource_name")
+
+		resource, found, err := pipeline.Resource(resourceName)
+		if err != nil {
+			logger.Error("failed-to-get-resource", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		var version atc.VersionDeleteBody
+		err = json.NewDecoder(r.Body).Decode(&version)
```

Comment:
We can actually move this code for decoding of the version to before we try to find a resource from the pipeline on line 20 https://github.com/concourse/concourse/blob/5db952798af1631e9d6f6396c2151dfbdbc005d8/atc/api/resourceserver/clear_resource_cache.go#L20 

That way the checking if the resource is not found https://github.com/concourse/concourse/blob/5db952798af1631e9d6f6396c2151dfbdbc005d8/atc/api/resourceserver/clear_resource_cache.go#L38-L51 is not split off from where the found boolean is set https://github.com/concourse/concourse/blob/5db952798af1631e9d6f6396c2151dfbdbc005d8/atc/api/resourceserver/clear_resource_cache.go#L20